walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -22,52 +22,68 @@
     List<Event> optionalEvents = new ArrayList<>();  
 
     for(Event event: events) {
-      int type = eventType(event, request);
-      if (type == 1) {
+      if (hasRequiredAttendees(event, request)) {
         requiredEvents.add(event);  
       }
-      else if (type == 2) {
+      //Only check for optional attendees if the event has no required one
+      else if (hasOptionalAttendees(event, request)) {
         optionalEvents.add(event);    
       }
     }
 
-    Collection<TimeRange> requiredTimes = getRequiredTime(requiredEvents, request.getDuration());
-    Collection<TimeRange> optionalTimes = getOptionalTime(optionalEvents, request.getDuration(), requiredTimes);
+    List<TimeRange> requiredTimes = getRequiredTime(requiredEvents, request.getDuration());
```

Comment:
Out of curiosity, what was the motivation to change from `Collection` to `List` everywhere?

Reply:
The sort and binarySearch methods called in getTimeAlternative do not work on `Collection`, but do on `List`